src/main/java/mat/client/umls/ManageUmlsPresenter.java

```diff
@@ -19,253 +19,261 @@
+                display.getErrorMessageDisplay().setMessage(MatContext.get().getMessageDelegate().getGenericErrorMessage());
+            }
+
+            @Override
+            public void onSuccess(final Void result) {
+            }
+        });
+    }
+
+    /**
+     * Method to clear all messages and un set user id and password input box's.
+     **/
+    private void resetWidget() {
+        display.getErrorMessageDisplay().clear();
+        display.getSuccessMessageAlert().clear();
+        display.getExternalLinkDisclaimer().setVisible(false);
+        display.getPasswordInput().setText("");
+        display.getUserIdText().setText("");
+        display.getUserIdGroup().setValidationState(ValidationState.NONE);
+        display.getPasswordGroup().setValidationState(ValidationState.NONE);
+        display.getMessageFormGrp().setValidationState(ValidationState.NONE);
+        display.getHelpBlock().setText("");
+    }
+
+    /**
+     * Private submit method - Calls to VSAC service.
+     **/
+    private void submit() {
+        display.getErrorMessageDisplay().clear();
+        display.getSuccessMessageAlert().clearAlert();
+        display.getExternalLinkDisclaimer().setVisible(false);
+
+        if (display.getUserIdText().getText().isEmpty()) {
+            display.getUserIdGroup().setValidationState(ValidationState.ERROR);
+            display.getHelpBlock().setIconType(IconType.EXCLAMATION_CIRCLE);
+            display.getHelpBlock().setText(MatContext.get().getMessageDelegate().getLoginUserRequiredMessage());
+            display.getMessageFormGrp().setValidationState(ValidationState.ERROR);
+            display.getSubmit().setEnabled(true);
+        } else if (display.getPasswordInput().getText().isEmpty()) {
+            display.getHelpBlock().setIconType(IconType.EXCLAMATION_CIRCLE);
+            display.getHelpBlock().setText(MatContext.get().getMessageDelegate().getPasswordRequiredMessage());
+            display.getMessageFormGrp().setValidationState(ValidationState.ERROR);
+            display.getUserIdGroup().setValidationState(ValidationState.SUCCESS);
+            display.getPasswordGroup().setValidationState(ValidationState.ERROR);
+            display.getSubmit().setEnabled(true);
+        } else {
+            display.getUserIdGroup().setValidationState(ValidationState.SUCCESS);
+            display.getPasswordGroup().setValidationState(ValidationState.SUCCESS);
+            vsacapiService.validateVsacUser(display.getUserIdText().getText(),
+                    display.getPasswordInput().getText(), new AsyncCallback<Boolean>() {
+                        @Override
+                        public void onFailure(final Throwable caught) {
+                            display.getErrorMessageDisplay().setMessage(
+                                    MatContext.get().getMessageDelegate().getUML_LOGIN_UNAVAILABLE());
+                            caught.printStackTrace();

```

Comment:
do we need to log this `printStackTrace()` ?

Reply:
well, i can switch to logging instead. I just re-formatted that code